.pipelines/ci.yml

```diff
@@ -36,161 +36,254 @@ variables:
+
+      - job: Lint_Az_ARO_Extension_1
+        dependsOn: Build_Test_And_Push_Az_ARO_Extension
+        pool:
+          name: 1es-aro-ci-pool
+        variables:
+          HOME: $(Agent.BuildDirectory)
+        steps:
+          - template: ./templates/template-checkout.yml
+          - script: |
+              set -xe
+              export AZDEV_CONFIG_DIR=$(Agent.BuildDirectory)/azdev-config
+              make test-python
+              [[ -z "$(git status -s)" ]]
+            target: python
+
+      - job: Build_Lint_Test_RP_And_Portal
+        dependsOn: Build_Test_And_Push_Az_ARO_Extension
+        pool:
+          name: 1es-aro-ci-pool
+        steps:
+          - template: ./templates/template-checkout.yml
+          - script: |
+              set -xe
+              export NO_CACHE=true
+
+              # Install podman
+              sudo rpm -ivh https://dl.fedoraproject.org/pub/epel-release-latest-8.noarch.rpm
+              sudo dnf install -y podman make
+              which podman
+              podman --version
+              whoami
+
+              # Start podman service
+              . ./hack/e2e/utils.sh
+              run_podman
+              validate_podman_running
+
+              # Run the CI process (build, test, coverage extraction, etc.)
+              export PODMAN_REMOTE_ARGS="-r --url=tcp://localhost:8888"
+              make ci-rp  # This step includes building, testing, and generating the coverage report
+
+              # Stop podman
+              kill_podman
+            target: ubi8
+
+      - job: Extract_Coverage_Report_2
+        dependsOn: Build_Lint_Test_RP_And_Portal  # Ensures that the image is built before this job
+        pool:
+          vmImage: 'ubuntu-latest'
+        variables:
+          ACR_FQDN: 'arosvcdev.azurecr.io'  # ACR FQDN
+          REPOSITORY: 'my-image-repo'
+          IMAGE_TAG: $(Build.SourceBranchName)
+        steps:
+          - checkout: self
+
+          # Authenticate with ACR using the Managed Identity service connection
+          - task: AzureCLI@2
+            inputs:
+              azureSubscription: 'ado-pipeline-dev-image-push'  # Service connection name
+              scriptType: 'bash'
+              scriptLocation: 'inlineScript'
+              inlineScript: |
+                az acr login --name ${ACR_FQDN%%.*}
+                docker login ${ACR_FQDN} -u "00000000-0000-0000-0000-000000000000" -p "$(az acr login --name ${ACR_FQDN%%.*} --expose-token --output tsv --query accessToken)"
+
+          - script: |
+              set -eux
+              # Extract the coverage report from the ACR-pushed image
+              docker create --name coverage-container ${ACR_FQDN}/${REPOSITORY}:${IMAGE_TAG}
+              docker cp coverage-container:/go/src/ARO-RP/cover.out $(Build.SourcesDirectory)/cover.out || {
+                  echo "Coverage file not found in container."
+                  exit 1
+              }
+              docker rm coverage-container
```

Comment:
can we move this logic into the Makefile `ci-rp` target, so that we can also view this file on local builds?